routes/auth.js

```diff
@@ -0,0 +1,100 @@
+var express = require('express');
+var router = express.Router();
+var utils = require('../utils.js');
+var ldapUtils = require('../ldap.js');
+
+module.exports = function(connection) {
+
+    /* Get all auth rules */
+    router.get('/auth', function (req, res, next) {
+
+        connection.query('select * from auth', function (error, rows, field) {
+
+            if(!!error){
+                console.log(error);
+            }
+            else {
+                res.setHeader('Content-Type', 'application/json');
+                res.json(rows);
+            }
+        })
+    });
+    
+    /* Create new auth rule */
+    router.post('/auth', utils.isAuthenticated, function (req, res, next) {
+        var atlas_id=req.body.atlas_id;
+
+        connection.query('SELECT * FROM atlas WHERE id=?', [atlas_id], function (error, rows, field) {
+            if(!!error){
+                console.log(error);
+            } else if(rows[0].created_by == req.session.user.uid || req.session.user.admin) {
+
+                var principal=req.body.principal;
+                ldapUtils.getUser(principal, function(error, user) {
+
+                    if(error) {
+                        console.log(error);
+                    } else if (user) {
+                        var token=req.body.token;
+                        if(token) {
+                            token = utils.hashToken(token);
+                        }
+                        var privileges=req.body.privileges;
+                        var expires=req.body.expires;
+                        connection.query('INSERT INTO auth(atlas_id,principal,token,privileges,expires) VALUES(?,?,?,?,?)', [atlas_id,principal,token,privileges,expires], function (error, rows, field) {
+                            if(!!error){
+                                console.log(error);
+                            } else {
+                                res.setHeader('Content-Type', 'application/json');
+                                var json = req.body;
+                                json.id = rows.insertId;
+                                res.json(json);
+                            }
+                        });        
+                    } else {
+                        res.status(400).send({ message: principal + ' not found' });
+                    }
+                }); 
+            } else {
+                res.send(401);
+            }
+        });
+    });
+
+    /* Delete auth rule with given id */
+    router.delete('/auth/:id', utils.isAuthenticated, function(req, res, next) {
+
+        var id=req.params['id'];
```

Comment:
If `id` is missing (empty/blank), return a `400 Bad Request` instead of trying to finding an entry in auth with id=''.